### PR TITLE
Upgrade PartitionedArrays from 0.3.x to 0.5.x

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

File: src/HYPRE.jl [hunk 1]
```diff
@@ -344,36 +343,49 @@ function subarray_unsafe_supported()
...
+    ghost_to_global_col = ghost_to_global(c)
+
+    # HYPRE requires contiguous row indices
+    ilower = own_to_global_row[1]
+    iupper = own_to_global_row[end]
+    @assert iupper - ilower + 1 == length(own_to_global_row)
+
+    # Extract sparse matrices from the SplitMatrix. We are only interested in the owned
+    # rows, so only consider own-own and own-ghost blocks.
+    Aoo = A.blocks.own_own::SparseMatrixCSC
```
Comment (fredrikekre) [inline src/HYPRE.jl:361]: @fverdugo is using `A.block.own_own` the correct way or is there some function to extract this part? I saw https://github.com/fverdugo/PartitionedArrays.jl/blob/b211949cc5842f3d6d99f8452b9f0233aef963a5/src/p_sparse_matrix.jl#L670-L672 but not sure what to pass as the second and third argument (although they are unused so doesn't matter...)

**Review by fredrikekre — COMMENTED** [review]
Also wanted to ask if using `.row_partition` and `.col_partition` like in https://github.com/fredrikekre/HYPRE.jl/blob/5bc27460f69278f02668971306c2455b41ae2fee/src/HYPRE.jl#L516 is the correct way?

**Comment by fredrikekre** [comment]
> Also wanted to ask if using .row_partition and .col_partition like in 

https://github.com/fverdugo/PartitionedArrays.jl/blob/b211949cc5842f3d6d99f8452b9f0233aef963a5/src/p_sparse_matrix.jl#L993 looks close. `partition(axes(A, 1))` perhaps? Would it make sense to have something like `local_axes(A)`?

**Comment by fverdugo** [comment]
Nice to see this merged so quickly!

> Also wanted to ask if using .row_partition and .col_partition like in

These fields are documented (thus public), but I always use `partition(axes(A,x))` outside PAs.

NB. AFAIK, Petsc (and provably Hypre) assumes that the global row ids are partitioned using a "block" partition (consecutive rows go to the same rank) and blocks are mapped to ranks in ascending rank id. Moreover ghost columns are sorted according to global id. None of these restrictions apply to PAs, which makes challenging to convert a PSparseMatrix to a Petsc matrix since you need to reformat data to match Petsc assumptions.

Take a look to this function to check how I am building a Petsc matrix from a PSparseMatrix:

https://github.com/fverdugo/PetscCall.jl/blob/3602086520882797b0210df22fabdb5af9a46a5a/src/ksp.jl#L246

You provably need to do something similar  (unless you are already doing it) so that HYPRE.jl is able to work with any PSparseMatrix.

**Comment by fredrikekre** [comment]
Yea, right now I am just throwing an error instead (https://github.com/fredrikekre/HYPRE.jl/blob/10bdaec7230a90666c0672f74b2587cca90e8797/ext/HYPREPartitionedArrays.jl#L36).

**Pull request merged**